tests/integration/targets/ndo_fex_device/tasks/main.yml

```diff
@@ -0,0 +1,196 @@
+# Test code for the MSO modules
+# Copyright: (c) 2025, Shreyas Srish (@shrsr) <ssrish@cisco.com>
+
+# GNU General Public License v3.0+ (see LICENSE or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+- name: Test that we have an ACI MultiSite host, username and password
+  ansible.builtin.fail:
+    msg: "Please define the following variables: mso_hostname, mso_username and mso_password."
+  when: mso_hostname is not defined or mso_username is not defined or mso_password is not defined
+
+# CLEAN ENVIRONMENT
+- name: Set vars
+  ansible.builtin.set_fact:
+    mso_info: &mso_info
+      host: "{{ mso_hostname }}"
+      username: "{{ mso_username }}"
+      password: "{{ mso_password }}"
+      validate_certs: "{{ mso_validate_certs | default(false) }}"
+      use_ssl: "{{ mso_use_ssl | default(true) }}"
+      use_proxy: "{{ mso_use_proxy | default(true) }}"
+      output_level: '{{ mso_output_level | default("debug") }}'
+
+# QUERY VERSION
+- name: Query mso version
+  cisco.mso.mso_version:
+    <<: *mso_info
+    state: query
+  register: version
+
+- name: Execute tasks only for FEX version > 2.0
+  when: version.current.version is version('2.0', '>')
+  block:
+
+    # SETUP
+    - name: Ensure Fabric Resource template does not exist
+      cisco.mso.ndo_template: &ndo_fabric_resource_template_absent
+        <<: *mso_info
+        template: ansible_fabric_resource_template
+        type: fabric_resource
+        state: absent
+
+    - name: Create a new Fabric Resource template
+      cisco.mso.ndo_template:
+        <<: *ndo_fabric_resource_template_absent
+        state: present
+      register: create_fabric_resource
+
+    # CREATE
+    - name: Create FEX Device (check mode)
+      cisco.mso.ndo_fex_device: &add_fex_device
+        <<: *mso_info
+        template: ansible_fabric_resource_template
+        name: ansible_fex_device
+        fex_id: 101
+        nodes: [203]
+        interfaces: ['1/3']
+        description: Ansible FEX Device
+        state: present
+      check_mode: true
+      register: cm_fex_device
+
+    - name: Create FEX Device
```

Comment:
Looks like the test is missing the idempotency checks? Create again, Update again and Delete again?